charts/skypilot/templates/rbac.yaml

```
@@ -15,59 +16,57 @@ metadata:
   name: {{ .Release.Name }}-api-role
 rules:
 {{ toYaml .Values.rbac.clusterRules | indent 2 }}
+{{- if .Values.rbac.manageRbacPolicies }}
+  - apiGroups: [ "rbac.authorization.k8s.io" ]
+    resources: [ "clusterroles", "clusterrolebindings" ]
+    verbs: [ "*" ]
+{{- end }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: {{ .Release.Name }}-api-role-binding
 subjects:
 - kind: ServiceAccount
-  name: {{ .Release.Name }}-api-sa
+  name: {{ include "skypilot.serviceAccountName" . }}
   namespace: {{ .Release.Namespace }}
 roleRef:
   kind: ClusterRole
   name: {{ .Release.Name }}-api-role
   apiGroup: rbac.authorization.k8s.io
 ---
-{{- $namespaces := list }}
-{{- range .Values.rbac.grantedNamespaces }}
-{{- $namespaces = append $namespaces .namespace }}
-{{- end }}
-{{- $namespaces = append $namespaces .Release.Namespace | uniq }}
-{{- range $namespaces }}
-{{/* Create namespaces in advance to enable RBAC rules, filter out the release namespace in case user accidentally added it to grantedNamespaces*/}}
-{{- if and (ne . $.Release.Namespace) (not (lookup "v1" "Namespace" "" .)) }}
----
-apiVersion: v1
-kind: Namespace
-metadata:
-  name: {{ . }}
-  annotations:
-    {{/* Keep the namespace when uninstalling the chart, so that the deployed sky resources (if any) can still work even if the API server get uninstalled */}}
-    helm.sh/resource-policy: keep
-{{- end }}
+{{- $namespace := .Values.kubernetesCredentials.inclusterNamespace | default .Release.Namespace }}
+{{- if ne $namespace .Release.Namespace -}}
+{{- include "skypilot.ensureNamespace" $namespace }}
+{{- end -}}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
-  name: {{ $.Release.Name }}-api-role
-  namespace: {{ . }}
+  name: {{ .Release.Name }}-api-role
+  namespace: {{ $namespace }}
 rules:
-{{ toYaml $.Values.rbac.namespaceRules | indent 2 }}
+{{ toYaml .Values.rbac.namespaceRules | indent 2 }}
+{{- if .Values.rbac.manageRbacPolicies }}
+  - apiGroups: [ "rbac.authorization.k8s.io" ]
+    resources: [ "roles", "rolebindings" ]
+    verbs: [ "*" ]
+  - apiGroups: [ "" ]
+    resources: [ "serviceaccounts" ]
+    verbs: [ "*" ]
+{{- end }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: {{ $.Release.Name }}-api-role-binding
-  namespace: {{ . }}
+  name: {{ .Release.Name }}-api-role-binding
+  namespace: {{ $namespace }}
 subjects:
 - kind: ServiceAccount
-  name: {{ $.Release.Name }}-api-sa
-  namespace: {{ $.Release.Namespace }}
+  name: {{ include "skypilot.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
```

Should this be $namespace?
```suggestion
  namespace: {{ $namespace }}
```

---

No, the service account should always be created in the same namespace with API server pod, while the permissions from other namespaces can be granted to it later